spark/src/main/java/com/netflix/iceberg/spark/source/Reader.java

```diff
@@ -385,8 +416,9 @@ private static UnsafeProjection projection(Schema finalSchema, Schema readSchema
     }
 
     private Iterator<InternalRow> open(FileScanTask task, Schema readSchema) {
-      InputFile location = fileIo.newInputFile(task.file().path().toString());
       CloseableIterable<InternalRow> iter;
+      InputFile location = inputFiles.get(task.file().path().toString());
+      Preconditions.checkNotNull(location, "Could not find InputFile associated with FileScanTask");
```

Comment:
The `location` assignment was reloacated after `iter`. Is that a requirement of some style enforcement? (That would be a good reason for the change.)

Usually, I try to avoid such changes to avoid git conflicts. Relocations needlessly increase the number of adjacent lines that might conflict when cherry-picking or rebasing.